hyperspy/misc/eels/tools.py

```diff
@@ -307,3 +308,61 @@ def eels_constant(s, zlp, t):
         data=(t * i0 / (332.5 * ke)) * np.log(1 + (beta * tgt / eaxis) ** 2))
     k.metadata.General.title = "EELS proportionality constant K"
     return k
+
+
+def get_edge_onset(data, start, end, x_axis, percent_position):
+    """Get the onset energy for a core loss edge.
+
+    Parameters
+    ----------
+    data : NumPy array
+    x_axis : NumPy array
+        Same size as data
+    start, end : scalar
+        Region where the edge onset will be found, in x_axis values.
+    percent_position : float
+        At what fraction of the ELNES signal the onset energy will be
+        found. See EELSModel.estimate_and_set_coreloss_edge_onset for more
+        information.
+
+    Returns
+    -------
+    onset_energy : scalar
+
+    Examples
+    --------
+    >>> from hyperspy.misc.eels.tools import get_edge_onset
+    >>> s = hs.datasets.artificial_data.get_core_loss_eels_signal()
+    >>> data, x_axis = s.data, s.axes_manager[0].axis
+    >>> onset = get_edge_onset(data, 600, 700, x_axis, 0.1)
+
+    """
+    if not ((percent_position >= 0.) and (percent_position <= 1.)):
+        raise ValueError(
+                "percent position ({0}) must be between 0 and 1".format(
+                    percent_position))
+    start_i = np.argmax(x_axis > start)
+    end_i = np.argmax(x_axis > end)
+    if end_i == 0:
+        end_i = len(x_axis)
```

Comment:
Wouldn't it be better to raise a ``ValueError`` here?

Reply:
Fixed by adding several `if`s at the beginning of the function, to check if `start`/`end`/+++ is out of bounds of x-axis.